resteasy/src/main/java/com/linecorp/armeria/client/resteasy/ArmeriaResteasyClientBuilder.java

```diff
@@ -0,0 +1,541 @@
+        if (connectionIdleTimeout > 0) {
+            clientFactoryBuilder.idleTimeoutMillis(connectionIdleTimeout);
+        }
+
+        final String defaultProxyHost = delegate.getDefaultProxyHostname();
+        if (defaultProxyHost != null) {
+            final int defaultProxyPort = delegate.getDefaultProxyPort();
+            final String defaultProxyScheme = delegate.getDefaultProxyScheme();
+            final SessionProtocol proxyProtocol =
+                    defaultProxyScheme == null ? SessionProtocol.HTTP
+                                               : SessionProtocol.of(defaultProxyScheme);
+            final InetSocketAddress defaultProxyAddress =
+                    InetSocketAddress.createUnresolved(defaultProxyHost,
+                                                       defaultProxyPort > 0 ? defaultProxyPort
+                                                                            : proxyProtocol.defaultPort());
+            clientFactoryBuilder.proxyConfig(ProxyConfig.connect(defaultProxyAddress, proxyProtocol.isTls()));
+        }
+
+        if (delegate.isTrustManagerDisabled()) {
+            clientFactoryBuilder.tlsNoVerify();
+        }
+        final KeyStore trustStore = delegate.getTrustStore();
+        final KeyStore keyStore = delegate.getKeyStore();
+        if (trustStore != null || keyStore != null) {
+            clientFactoryBuilder.tlsCustomizer(sslContextBuilder ->
+                                                       SslContextConfigurator.configureSslContext(
+                                                               sslContextBuilder,
+                                                               keyStore,
+                                                               delegate.getKeyStorePassword(),
+                                                               trustStore,
+                                                               delegate.isTrustSelfSignedCertificates())
+            );
+        }
+        webClientBuilder.factory(clientFactoryBuilder.build());
+        return webClientBuilder.build();
+    }
+
+    private ResteasyClient build(WebClient webClient) {
+        final long readTimeout = delegate.getReadTimeout(TimeUnit.MILLISECONDS);
+        final int bufferSize = delegate.getResponseBufferSize();
+        final ClientHttpEngine armeriaEngine = new ArmeriaJaxrsClientEngine(
+                webClient,  bufferSize > 0 ? bufferSize : ArmeriaJaxrsClientEngine.DEFAULT_BUFFER_SIZE,
+                readTimeout > 0 ? Duration.ofMillis(readTimeout) : null);
+        return delegate.httpEngine(armeriaEngine).build();
+    }
+
+    @Override
+    public ResteasyClient build() {
+        return build(buildWebClient());
+    }
+
+    /**
+     * {@link ArmeriaJaxrsClientEngine} will always be set as an {@link ClientHttpEngine}.
+     * @throws UnsupportedOperationException always
+     */
+    @Override
+    public ResteasyClientBuilder httpEngine(ClientHttpEngine httpEngine) {
```

Comment:
I think we need to change the return type to `ArmeriaResteasyClientBuilder` for all setter in this builder.